drizzlepac/hlautils/catalog_utils.py

```diff
@@ -317,8 +331,45 @@ def __init__(self, image, param_dict, debug, tp_sources):
 
         self.sourcelist_filename = self.imgname.replace(self.imgname[-9:], self.catalog_suffix)
 
-        # Initialize attributes which get computed by class methods
-        self.bkg_used = None  # actual background used for source identification/measurement
+        # Compute catalog-independent attributes to be used for photometry computations
+        # Compute AB mag zeropoint
+        photplam = self.image.imghdu[1].header['photplam']
+        photflam = self.image.imghdu[1].header['photflam']
+        # FIX: replace the constants
+        self.ab_zeropoint = -2.5 * np.log10(photflam) - 21.10 - 5.0 * np.log10(photplam) + 18.6921
+
+        # Compute average gain
+        # FIX not four amps for all detectors
+        self.gain = self.image.imghdu[0].header['exptime'] * np.mean([self.image.imghdu[0].header['atodgna'],
+                                                                      self.image.imghdu[0].header['atodgnb'],
+                                                                      self.image.imghdu[0].header['atodgnc'],
+                                                                      self.image.imghdu[0].header['atodgnd']])
```

Comment:
Use of "header['atodgn*']" would create a list of whatever values are present in the header and use of list comprehension can generate the final values from this list without explicitly looking for all 4 amps.  Logic could even be used with this to only include keyword values that are non-zero since observations which only use 1 amp at a time set all the other keyword values to 0.0, which would ruin the interpretation of the gain entirely.  See 'ibvz01010' as an example of where this is done (0.0, 0.0, 1.559999, 0.0).

Reply:
Replaced the hard-coded search and evaluation of the four gain header keywords by the above suggestion of searching for the keywords using wildcards and list comprehension!